frontend/src/components/nav/navbar.js

```diff
@@ -56,6 +57,15 @@ class NavBar extends React.Component {
         }
     }
 
+    debounceSearch(func, delay) {
+        let inDebounce;
+        return (e) => {
+            e.persist();
+            clearTimeout(inDebounce);
+            inDebounce = setTimeout(func.bind(this, e), delay);
```

Comment:
window.setTimeout should accept an anonymous function that will reset timerId when func executes.

window.setTimeout(() => {
  func(e);
  timerId = null;
}, delay);